includes/abstracts/abstract-wc-gateway-ppec.php

```diff
@@ -67,6 +67,23 @@ public function __construct() {
 		} else {
 			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_scripts' ) );
 		}
+
+		add_filter( 'woocommerce_ajax_get_endpoint', array( $this, 'pass_return_args_to_ajax' ), 10, 2 );
+	}
+
+	/**
+	 * Pass woo return args to AJAX endpoint when the checkout updates from the frontend
+	 * so that the order button gets set correctly.
+	 *
+	 * @param  string $request Optional.
+	 * @return string
+	 */
+	public function pass_return_args_to_ajax( $request ) {
+		if ( isset( $_GET['woo-paypal-return'] ) ) {
+			$request .= '&woo-paypal-return=1';
```

Comment:
Can't use `add_query_arg` on `$request` here because it will automatically encode the percents in `wc-ajax=%%endpoint%%`.